tmt/steps/prepare/__init__.py

```diff
@@ -53,8 +53,13 @@ def go(self):
 
         packages = []
         for step in self.plan.steps():
-            if getattr(step, 'requires', False):
-                packages += step.requires()
+            if hasattr(step, 'steps'):
+                for inner_step in step.steps:
+                    if getattr(inner_step, 'requires', False):
+                        packages += inner_step.requires()
```

Comment:
That's strange. This shouldn't be the case... let me test+fix in #159, please.

Reply:
@psss any idea why do we need `for inner_step in step.steps:`?

Reply:
I think the `requires` method should be called only for `discover` (gathered L1 require) and `execute` (test framework execution requires like `beakerlib`) steps without touching inner data.

Reply:
I agree, @lukaszachy why is the original implementation an issue for you?